File: solutions/csharp/lucians-luscious-lasagna/1/LuciansLusciousLasagna.cs
public class Lasagna
{
    private const int MinutesPerLayer = 2;

    public int ExpectedMinutesInOven() => 40;

    public int RemainingMinutesInOven(int minutesBeenInOven)
        => ExpectedMinutesInOven() - minutesBeenInOven;

    public int PreparationTimeInMinutes(int layers)
        => layers * MinutesPerLayer;

    public int ElapsedTimeInMinutes(int layers, int minutesBeenInOven)
        => PreparationTimeInMinutes(layers) + minutesBeenInOven;
}

File: solutions/fsharp/lucians-luscious-lasagna/1/LuciansLusciousLasagna.fs
module LuciansLusciousLasagna

let expectedMinutesInOven = 40

let remainingMinutesInOven currentMinutesInOven =
    expectedMinutesInOven - currentMinutesInOven

let preparationTimeInMinutes numberOfLayers = numberOfLayers * 2

let elapsedTimeInMinutes numberOfLayers currentMinutesInOven =
    let preparationTime = preparationTimeInMinutes numberOfLayers
    let ovenTime = remainingMinutesInOven currentMinutesInOven
    preparationTime + expectedMinutesInOven - ovenTime
